.github/workflows/test-integrations.yml

```diff
@@ -177,7 +177,7 @@ jobs:
       contents: read
     strategy:
       matrix:
-        vault-version: ["1.16.2", "1.15.6", "1.14.10"]
+        vault-version: ["1.17.5", "1.16.9", "1.15.14"]
```

Comment:
Looking at https://releases.hashicorp.com/vault, I think the latest CE release for 1.15 was 1.15.6 - that's why the download of `https://releases.hashicorp.com/vault/1.15.14/vault_1.15.14_linux_amd64.zip` (doesn't include`+ent`, so it's not a valid download URL) is failing. Similar story for Nomad: https://releases.hashicorp.com/nomad.

This is similar to the issue we ran into w/ `consul-k8s` acceptance tests once `consul` CE moved out of maintenance for all but the latest CE version.

Our options here are to either:
- Use the latest CE versions
- Start pulling Ent versions and configure licenses as secrets in CI to test them

The downside of the second option is that any community PRs against `consul` (CE) will fail integration tests, requiring an employee w/ repo write access to run them in order to gain access to the license secrets.

The way around this, and the approach I'd probably recommend, is testing CE versions in CE, and Ent versions in Ent. That would create a cleaner separation, but take a bit more time since we'll need to reconfigure each repo's jobs to target different release binaries. Shouldn't be too much work, but probably worth filing a ticket and putting some heads together since it won't be a simple bump anymore.

For now, we can go ahead and update to the latest CE versions:
- Vault: 1.17.5, 1.16.3, 1.15.6
- Nomad: 1.8.3, 1.7.7, 1.6.10